server/routes/makeAReferral/completion-deadline/completionDeadlineForm.ts

```diff
@@ -1,20 +1,86 @@
 import { Request } from 'express'
-import DraftReferral from '../../../models/draftReferral'
+import * as ExpressValidator from 'express-validator'
 import errorMessages from '../../../utils/errorMessages'
 import CalendarDayInput from '../../../utils/forms/inputs/calendarDayInput'
 import { FormData } from '../../../utils/forms/formData'
+import { ReferralDetailsFormUpdate } from '../../../models/referralUpdate'
+import { FormValidationError } from '../../../utils/formValidationError'
+import FormUtils from '../../../utils/formUtils'
+import { FormValidationResult } from '../../../utils/forms/formValidationResult'
 
 export default class CompletionDeadlineForm {
-  constructor(private readonly request: Request) {}
+  constructor(private readonly request: Request, private readonly isSentReferral: boolean) {}
 
-  async data(): Promise<FormData<Partial<DraftReferral>>> {
+  static readonly completionDeadlineFieldId = 'completion-deadline'
+
+  static readonly reasonForChangeFieldId = 'reason-for-change'
+
+  async data(): Promise<FormData<ReferralDetailsFormUpdate>> {
     const input = new CalendarDayInput(this.request, 'completion-deadline', errorMessages.completionDeadline)
-    const result = await input.validate()
+    const completionDeadlineResult = await input.validate()
+    let reasonForChangeResult: FormValidationResult<string> | null = null
+
+    if (this.isSentReferral) {
+      reasonForChangeResult = await CompletionDeadlineForm.validate(this.request)
+
+      return completionDeadlineResult.error || reasonForChangeResult!.error
+        ? {
+            paramsForUpdate: null,
+            error: {
+              errors: [
+                ...(completionDeadlineResult.error?.errors ?? []),
+                ...(reasonForChangeResult!.error?.errors ?? []),
+              ],
+            },
+          }
+        : {
+            error: null,
+            paramsForUpdate: {
+              draftReferral: { completionDeadline: completionDeadlineResult.value.iso8601 },
+              reasonForChange: this.request.body[CompletionDeadlineForm.reasonForChangeFieldId],
+            },
+          }
+    }
+    return completionDeadlineResult.value
+      ? {
+          error: null,
+          paramsForUpdate: {
+            draftReferral: { completionDeadline: completionDeadlineResult.value.iso8601 },
+            reasonForChange: null,
+          },
+        }
+      : { error: completionDeadlineResult.error, paramsForUpdate: null }
+  }
 
-    if (result.value) {
-      return { error: null, paramsForUpdate: { completionDeadline: result.value.iso8601 } }
+  static get validations() {
+    return [
+      ExpressValidator.body(CompletionDeadlineForm.reasonForChangeFieldId)
+        .notEmpty({ ignore_whitespace: true })
+        .withMessage(errorMessages.completionDeadline.cannotBeEmpty),
```

Comment:
this variable name suggests that the message is for the scenario when the completion deadline field is empty - it is actually for then the reason for change field is empty (i think?)